src/maker/mod.rs

```
@@ -104,39 +205,48 @@ pub async fn start_maker_server(maker: Arc<Maker>) -> Result<(), MakerError> {
         }
     }
 
-    let listener = TcpListener::bind((Ipv4Addr::LOCALHOST, maker.config.port)).await?;
-    log::info!("Listening On Port {}", maker.config.port);
+    let mut directory_onion_address = maker.config.directory_server_onion_address.clone();
 
-    let (server_loop_comms_tx, mut server_loop_comms_rx) = mpsc::channel::<MakerError>(100);
-    let mut accepting_clients = true;
-    let mut last_rpc_ping = Instant::now();
-    let mut last_directory_servers_refresh = Instant::now();
+    if cfg!(feature = "integration-test") {
+        let directory_hs_path_str = "/tmp/tor-rust-directory/hs-dir/hostname".to_string();
+        let directory_hs_path = PathBuf::from(directory_hs_path_str);
+        let mut directory_file = fs::File::open(directory_hs_path).unwrap();
+        let mut directory_onion_addr: String = String::new();
+        directory_file
+            .read_to_string(&mut directory_onion_addr)
+            .unwrap();
+        directory_onion_addr.pop();
+        directory_onion_address = format!("{}:{}", directory_onion_addr, 8080);
+    }
 
-    let maker_clone_1 = maker.clone();
-    std::thread::spawn(move || {
-        log::info!(
-            "[{}] Spawning Connection status check thread",
-            maker_clone_1.config.port
-        );
-        check_for_idle_states(maker_clone_1).unwrap();
-    });
+    let address = directory_onion_address.as_str();
 
-    let maker_clone_2 = maker.clone();
-    std::thread::spawn(move || {
-        log::info!(
-            "[{}] Spawning contract-watcher thread",
-            maker_clone_2.config.port
-        );
-        check_for_broadcasted_contracts(maker_clone_2).unwrap();
-    });
+    log::warn!("Directory onion address : {}", directory_onion_address);
+
+    let mut stream =
+        Socks5Stream::connect(format!("127.0.0.1:{}", maker_socks_port).as_str(), address)
+            .await
+            .map_err(|_e| MakerError::General("Error with socks "))?
+            .into_inner();
```

Can we propagate the error instead of mapping this? would be useful..